pallets/nft/src/lib.rs

```diff
@@ -423,6 +441,18 @@ impl<T: Config> Module<T> {
             PortfolioNFT::contains_key(&caller_portfolio, (&asset_id, &nft_id)),
             Error::<T>::NFTNotFound
         );
+        ensure!(
+            !PortfolioLockedNFT::contains_key(&caller_portfolio, (&asset_id, &nft_id)),
+            Error::<T>::NFTIsLocked
+        );
+
+        let actual_keys = CollectionKeys::get(collection_id).len();
```

Comment:
I think this would be more costly then using `MetadataValue::clear_prefix`.  Also we can check the results to see if all keys were removed (`result.maybe_cursor == None`).